lxd/api_internal.go

```diff
@@ -699,16 +699,29 @@ func internalImportFromBackup(s *state.State, projectName string, instName strin
 		return fmt.Errorf(`The storage pool's %q driver %q conflicts with the driver %q recorded in the instance's backup file`, instancePoolName, pool.Driver().Info().Name, backupConf.Pool.Driver)
 	}
 
+	// Apply the UUIDs that will be set in the database.
+	// srcBackup contains the already set UUIDs for both volumes and their snapshots.
+	// This is required down the line when comparing the actual snapshots on source with the ones from the config.
+	// Some storage drivers require the volume's UUID in order to generate the correct volume name.
+	backupConf.Volume.Config["volatile.uuid"] = srcBackup.Config.Volume.Config["volatile.uuid"]
```

Comment:
I dont really understand why this is needed,because the volatile.uuid of volumes and snapshots should already be in the backup.yaml.

Reply:
Removed as now obsolete due to https://github.com/canonical/lxd/pull/12904#discussion_r1492977454.